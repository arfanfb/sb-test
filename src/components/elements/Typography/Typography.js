import React from 'react';
import PropTypes from 'prop-types';
import styles from './Typography.module.scss';

export default function Typography(props) {
  const { 'class-name': classNameProps, tag: TagProp, variant, weight } = props;
  const className = [
    styles['typography'],
    styles[`${variant}`],
    styles[`${weight}`],
    classNameProps,
  ].join(' ');

  const preVariant = variant[0];
  const postVariant = variant.split('-')[1];
  const Tag = variant[0] === 'h' ? `${preVariant+postVariant}` : TagProp;
  return (
    ((preVariant === 'b' || preVariant === 'c')&& !TagProp)
      ? <label className={className} {...props} />
      : <Tag className={className} {...props} />
  );
}

Typography.defaultProps = {
  'class-name': undefined,
  tag: 'p',
  variant: 'body-1',
  weight: undefined,
};

Typography.propTypes = {
  'class-name': PropTypes.string,
  tag: PropTypes.string,
  variant: PropTypes.string,
  weight: PropTypes.string,
};
