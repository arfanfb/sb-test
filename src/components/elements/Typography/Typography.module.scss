.headline-1{
  font-size:60px;
  letter-spacing: -0.5px;
  font-weight: 800;
}
.headline-2{
  font-size:48px;
  letter-spacing: 0;
   font-weight: 800;
}
.headline-3{
  font-size:34px;
  letter-spacing: 0.25px;
  font-weight: 800;
}
.headline-4{
  font-size:24px;
  letter-spacing: 0;
  font-weight: 800;
}
.headline-5{
  font-size:20px;
  letter-spacing: 0.15px;
  font-weight: 600;
}
.subtitle-1{
  font-size: 16px;
  letter-spacing: 0.15px;
  font-weight: 700;
}
.subtitle-2{
  font-size: 14px;
  letter-spacing: 1;
  font-weight: 600;
}
.body-1{
  font-size: 16px;
  letter-spacing: 0.5px;
}
.body-2{
  font-size: 14px;
  letter-spacing: 0.25px;
}
.button{
  font-size:14px;
  letter-spacing: 1px;
  font-weight: 700;
}
.caption{
  font-size:12px;
  letter-spacing: 0.4px;
}
.overline-1{
  font-size: 10px;
  letter-spacing: 0.24px;
}
.overline-2{
  font-size: 10px;
  letter-spacing: 0.24px;
  font-weight: 600;
}
.extrabold{
  font-weight: 800;
}
.bold{
  font-weight:700
}
.semibold{
  font-weight: 600;
}
.medium{
  font-weight: 500;
}
.reguler{
  font-weight:400
}
.light{
  font-weight: 300;
}
.typography{
  font-style:normal;
}
