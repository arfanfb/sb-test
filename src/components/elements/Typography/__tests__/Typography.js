import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Typography from '../Typography';

describe('src/components/Typography', () => {
  it('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Typography />);
    expect(tree).toMatchSnapshot();

  });

  it('expect class-name style', () => {
    const result = Typography({
      ...Typography.defaultProps,
      classNameProps:'style',
    });
    const { classNameProps } = result.props;
    expect(classNameProps).toBe('style');
  });

  it('expect variant heading', () => {
    const result = Typography({
      ...Typography.defaultProps,
      variant:'heading 1',
    });
    const { variant } = result.props;
    expect(variant).toBe('heading 1');
  });

  it('expect variant body', () => {
    const result = Typography({
      ...Typography.defaultProps,
      variant:'body 1',
      tag: 'label',
    });
    const { tag } = result.props;
    expect(tag).toBe('label');
  });
});
