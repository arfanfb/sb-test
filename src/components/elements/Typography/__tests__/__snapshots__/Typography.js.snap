// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/Typography render 1`] = `
<p
  className="   "
  tag="p"
  variant="body-1"
/>
`;
